packages/web-components/fast-element/src/styles/element-styles.ts

```diff
@@ -92,53 +152,21 @@ export abstract class ElementStyles {
     }
 
     /**
-     * Create ElementStyles from ComposableStyles.
+     * Sets the strategy that handles adding/removing these styles for an element.
+     * @param strategy - The strategy to use.
      */
-    public static readonly create: ElementStyleFactory = (() => {
-        if (DOM.supportsAdoptedStyleSheets) {
-            const styleSheetCache = new Map();
-            return (styles: ComposableStyles[]) =>
-                // eslint-disable-next-line @typescript-eslint/no-use-before-define
-                new AdoptedStyleSheetsStyles(styles, styleSheetCache);
-        }
-
-        // eslint-disable-next-line @typescript-eslint/no-use-before-define
-        return (styles: ComposableStyles[]) => new StyleElementStyles(styles);
-    })();
-}
-
-function reduceStyles(
-    styles: ReadonlyArray<ComposableStyles>
-): (string | CSSStyleSheet)[] {
-    return styles
-        .map((x: ComposableStyles) =>
-            x instanceof ElementStyles ? reduceStyles(x.styles) : [x]
-        )
-        .reduce((prev: string[], curr: string[]) => prev.concat(curr), []);
-}
+    public withStrategy(Strategy: ConstructibleStyleStrategy): this {
+        this._strategy = new Strategy(reduceStyles(this.styles));
+        return this;
+    }
 
-function reduceBehaviors(
-    styles: ReadonlyArray<ComposableStyles>
-): ReadonlyArray<Behavior<HTMLElement>> | null {
-    return styles
-        .map((x: ComposableStyles) => (x instanceof ElementStyles ? x.behaviors : null))
-        .reduce(
-            (
-                prev: Behavior<HTMLElement>[] | null,
-                curr: Behavior<HTMLElement>[] | null
-            ) => {
-                if (curr === null) {
-                    return prev;
-                }
-
-                if (prev === null) {
-                    prev = [];
-                }
-
-                return prev.concat(curr);
-            },
-            null as Behavior<HTMLElement>[] | null
-        );
+    /**
+     * Sets the default strategy type to use when creating style strategies.
+     * @param Strategy - The strategy type to construct.
+     */
+    public static setDefaultStrategy(Strategy: ConstructibleStyleStrategy) {
```

Comment:
@nicholasrice This is the API that SSR will call to set its own strategy for handling add/remove of styles.